packages/edit-post/src/components/visual-editor/style.scss

```diff
@@ -35,9 +35,17 @@
 			height: 0; // This collapses the container to an invisible element without margin.
 			text-align: center;
 
+			// This float rule takes the toolbar out of the flow, without it having to be absolute positioned.
+			// This is necessary because otherwise the mere presence of the toolbar can push down content.
+			// Pairs with relative rule on line 49.
+			float: left;
```

Comment:
Didn't we remove this for another reason? Just want to make sure we're not regressing something else?

Reply:
We removed it because it was _one_ way to take the toolbar out of the flow of the post. But that method for taking it out of the flow became redundant with other refactors to the block toolbar, and so we took the rule out _partially_ because it was redundant, and partially to improve how float adjacent blocks handled.

For example you might have a left floated image with a paragraph after it. With the float rule on the toolbar intact, you'd get this:

<img width="973" alt="screenshot 2018-11-26 at 11 00 13" src="https://user-images.githubusercontent.com/1204802/49006943-8da49100-f16a-11e8-802c-fffaa9c14a95.png">

By removing the float rule, and instead applying a position: absolute (the one we override in that code above), we get this:

![floats](https://user-images.githubusercontent.com/1204802/49006961-9ac18000-f16a-11e8-83a5-9f2a5ce5cd47.gif)

So will this regress other things? No, it shouldn't, because it adds only a single rule that's scoped to wide and fullwide blocks, and since those types of blocks should always clear floats that preceed them, there shouldn't be situations where this float would mis-position the toolbar.

If we wanted to test, the test case would be to have a float followed by a wide image, or a float followed by a fullwide image, and then testing that the block toolbar looks right in both sitautions. Which in my testing it does.

Reply:
Thanks for the detailed explanation, that makes sense to me 👍

Reply:
👍 👍

It's a good question to be asked, an important one for deciding if/when to merge.